### ISSUE TITLE
SpringBootProperties_2_4 - Profile Group config gets corrupted if specific yml indenting

### ISSUE THREAD

**Issue by nmck257** [openrewrite/rewrite-spring#189]
`org.openrewrite.java.spring.boot2.SpringBootProperties_2_4` contains the following change:

```yml
  - org.openrewrite.properties.ChangePropertyKey:
      oldPropertyKey: spring.profiles
      newPropertyKey: spring.config.activate.on-profile
  - org.openrewrite.yaml.ChangePropertyKey:
      oldPropertyKey: spring.profiles
      newPropertyKey: spring.config.activate.on-profile
```

...which works as intended w/ Spring's changes to profile-specific documents in 2.4.

Spring Boot 2.4 also introduces Profile Groups, with the following syntax:

```properties
spring.profiles.group.prod=proddb,prodmq,prodmetrics
```

This works fine in most cases, but the recipe above will unintentionally break usages of that new syntax *if* it's in a yml file with indented hierarchy:

```kotlin
import org.junit.jupiter.api.Test
import org.openrewrite.Recipe
import org.openrewrite.test.RecipeSpec
import org.openrewrite.test.RewriteTest
import org.openrewrite.yaml.ChangePropertyKey

class MyTest : RewriteTest {
    override fun defaults(spec: RecipeSpec) {
        spec.recipe(ChangePropertyKey("spring.profiles", "spring.config.activate.on-profile", null, null))
    }

    @Test
    fun `does not alter spring profile group config on single-line`() = rewriteRun(yaml(
        """
            spring.profiles.group.prod: proddb,prodmq,prodmetrics
        """
    )) // success

    @Test
    fun `does not alter spring profile group config on multi-line`() = rewriteRun(yaml(
        """
            spring:
              profiles:
                group:
                  prod: proddb,prodmq,prodmetrics
        """
    )) // failure -- actual: spring.config.activate.on-profile.group.prod: proddb,prodmq,prodmetrics
}
```

Repos migrating into 2.4 for the first time will be fine, but existing post-2.4 repos who use this feature and receive this recipe will break (and given how UpgradeSpringBoot recipes are chained together, that's not unlikely to occur).

**Comment by nmck257** [comment]
(revised with a test case to demonstrate)

**Comment by nmck257** [comment]
Closed by #212
